oscarapi/views/basic.py

```diff
@@ -98,44 +108,57 @@ def get(self, request, pk=None, format=None):
         return Response(ser.data)
 
 
-class ProductAvailability(generics.RetrieveAPIView):
-    model = Product
-    serializer_class = serializers.ProductAvailabilitySerializer
+class ProductAvailability(APIView):
```

Comment:
Yeah, but my implementation of the serializer class was ugly so now at least this is just one way to do it. Could try to rewrite both get methods though.